src/Sarif/CodeGenHints.json

```diff
@@ -434,6 +442,30 @@
       }
     }
   ],
+  "threadFlowLocation": [
+    {
+      "kind": "BaseTypeHint",
+      "arguments": {
+        "baseTypeNames": [
+          "PropertyBagHolder"
+        ]
+      }
+    }
+  ],
+  "ThreadFlowLocation.State": [
+    {
+      "kind": "DictionaryHint"
+    }
+  ],
+  "ThreadFlowLocation.Importance": [
+    {
+      "kind": "EnumHint",
+      "arguments": {
+        "typeName": "ThreadFlowLocationImportance",
+        "description": "Values specifying the importance of an \"threadFlowLocation\" within the \"codeFlow\" in which it occurs"
+      }
```

Comment:
>} [](start = 6, length = 1)

Bug: As a result of this declaration, I would expect the generated code to include a new file Autogenerated\ThreadFlowLocationImportance.cs. (See Autogenerated\Baseline.cs for an example of what you'd expect.)

I don't know if it's missing because you neglected to omit it, or whether the code generator decided not to bother emitting it because you left out `arguments.memberNames`. Either way, it needs to be there.

Reply:
And I don't know why AppVeyor doesn't catch this. The AppVeyor build should break because `ThreadFlowLocation` declares a member `Importance` of type `ThreadFlowLocationImportance`, but that type isn't declared anywhere.

---
In reply to: [208047570](https://github.com/Microsoft/sarif-sdk/pull/989#discussion_r208047570) [](ancestors = 208047570)

Reply:
CodeGen is working here. I looked at jschema code, the enum member name is derived on line 83 of the following file:

[https://github.com/Microsoft/jschema/blob/master/src/Json.Schema.ToDotNet/EnumGenerator.cs](https://github.com/Microsoft/jschema/blob/master/src/Json.Schema.ToDotNet/EnumGenerator.cs)

---
In reply to: [208048947](https://github.com/Microsoft/sarif-sdk/pull/989#discussion_r208048947) [](ancestors = 208048947,208047570)